diff -ubr shaderc-2023.3.orig/libshaderc_util/src/compiler.cc shaderc-2023.3/libshaderc_util/src/compiler.cc
--- shaderc-2023.3.orig/libshaderc_util/src/compiler.cc	2023-03-18 00:56:41.000000000 +0900
+++ shaderc-2023.3/libshaderc_util/src/compiler.cc	2023-03-22 09:22:49.071978600 +0900
@@ -20,7 +20,7 @@
 #include <thread>
 #include <tuple>
 
-#include "SPIRV/GlslangToSpv.h"
+#include "glslang/SPIRV/GlslangToSpv.h"
 #include "libshaderc_util/format.h"
 #include "libshaderc_util/io_shaderc.h"
 #include "libshaderc_util/message.h"
shaderc-2023.3/libshaderc_util/src のみに存在: compiler.cc.orig
diff -ubr shaderc-2023.3.orig/libshaderc_util/src/resources.cc shaderc-2023.3/libshaderc_util/src/resources.cc
--- shaderc-2023.3.orig/libshaderc_util/src/resources.cc	2023-03-18 00:56:41.000000000 +0900
+++ shaderc-2023.3/libshaderc_util/src/resources.cc	2023-03-22 09:22:31.807364664 +0900
@@ -125,15 +125,6 @@
     /* .maxTaskWorkGroupSizeY_NV = */ 1,
     /* .maxTaskWorkGroupSizeZ_NV = */ 1,
     /* .maxMeshViewCountNV = */ 4,
-    /* .maxMeshOutputVerticesEXT = */ 256,
-    /* .maxMeshOutputPrimitivesEXT = */ 256,
-    /* .maxMeshWorkGroupSizeX_EXT = */ 128,
-    /* .maxMeshWorkGroupSizeY_EXT = */ 128,
-    /* .maxMeshWorkGroupSizeZ_EXT = */ 128,
-    /* .maxTaskWorkGroupSizeX_EXT = */ 128,
-    /* .maxTaskWorkGroupSizeY_EXT = */ 128,
-    /* .maxTaskWorkGroupSizeZ_EXT = */ 128,
-    /* .maxMeshViewCountEXT = */ 4,
     /* .maxDualSourceDrawBuffersEXT = */ 1,
     // This is the glslang TLimits structure.
     // It defines whether or not the following features are enabled.
